R/table-create.R

```diff
@@ -53,13 +57,22 @@ setMethod("sqlCreateTable", signature("DBIConnection"),
       fields <- vapply(fields, function(x) DBI::dbDataType(con, x), character(1))
     }
 
+    if (!is.null(pk)) {
+      stopifnot(all(pk %in% names(fields)))
+      nullable <- ifelse(names(fields) %in% pk, " NOT NULL", "")
+    } else {
+      nullable <- ""
+    }
+
     field_names <- dbQuoteIdentifier(con, names(fields))
     field_types <- unname(fields)
     fields <- paste0(field_names, " ", field_types)
 
     SQL(paste0(
       "CREATE ", if (temporary) "TEMPORARY ", "TABLE ", table, " (\n",
-      "  ", paste(fields, collapse = ",\n  "), "\n)\n"
+      "  ", paste0(fields, nullable, collapse = ",\n  "),
+      if (!is.null(pk)) paste0(",\n  PRIMARY KEY (", paste(dbQuoteIdentifier(con, pk), collapse = ", "), ")"),
```

Comment:
How standard is this syntax?

Reply:
Pretty much, I think, but I don't know. We'll find out soon enough, we can override in backends -- this is a method.